### PR TITLE
Add parser class basic functionality

### LINKED ISSUE #18: Create Parser class
To parser user commands.

### PR DESCRIPTION
**Pull request by ptejasv**
# Changelog
- add functionality to `Parser` class to parse all v1.0 commands in 
  `parser.parse()` as run from main now returns an executable Command object
- initialise all `Command` classes (execution functionality not added yet)

### CODE REVIEW

File: src/main/java/seedu/duke/command/AddLessonCommand.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package seedu.duke.command;
+
+public class AddLessonCommand extends Command {
+    private String lessonDescription;
```
Comment (richwill28) [inline src/main/java/seedu/duke/command/AddLessonCommand.java:4]: `title` should be a better naming for consistency with the `Lesson` class.

File: src/main/java/seedu/duke/command/AddTaskCommand.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package seedu.duke.command;
+
+public class AddTaskCommand extends Command {
+    private String taskDescription;
```
Comment (richwill28) [inline src/main/java/seedu/duke/command/AddTaskCommand.java:4]: Same issue as mentioned above.

File: src/main/java/seedu/duke/command/DeleteCommand.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package seedu.duke.command;
+
+import javax.swing.text.DefaultEditorKit;
```
Comment (richwill28) [inline src/main/java/seedu/duke/command/DeleteCommand.java:3]: Unused import?

File: src/main/java/seedu/duke/Duke.java [hunk 4]
```diff
@@ -10,17 +11,24 @@
...
     }
 
     public void startProgram() {
         boolean isExit = false;
         while (!isExit) {
             String userResponse = ui.readUserResponse();
+            try {
+                Command command = parser.parse(userResponse);
+            } catch (DukeException e) {
+                e.printStackTrace(); // error while parsing input
```
Comment (richwill28) [inline src/main/java/seedu/duke/Duke.java:30]: Do note that `printStackTrace` is not exactly user-friendly. For development purpose it's ok, but we will have to remove it by the end our production.

File: src/main/java/seedu/duke/parser/Parser.java [hunk 5]
```diff
@@ -1,4 +1,157 @@
...
+    public static final String LESSON_KEYWORD = "lesson";
+    public static final String ALL_KEYWORD = "all";
+    public static final String COMMAND_LIST = "list";
+    public static final String COMMAND_DONE = "done";
+    public static final String COMMAND_DELETE = "delete";
+    public static final String COMMAND_FIND = "find";
+    public static final String FLAG_DAY = "-d";
+    public static final String FLAG_INFORMATION = "-i";
+    public static final String FLAG_START_TIME = "-s";
+    public static final String FLAG_END_TIME = "-e";
```
Comment (richwill28) [inline src/main/java/seedu/duke/parser/Parser.java:18]: Consider refactoring these to `enum`, to reduce clutter on the `Parser` class. But we can do this at another time, doesn't have to be this PR.

**Review by richwill28 — APPROVED** [review]
Should be okay for now, but we will have to refactor the `Parser` class later on to improve SLAP. Also, try updating the `text-ui-test` and confirm whether everything is working correctly. This will resolve the GitHub CI checks.

File: src/main/java/seedu/duke/Duke.java (hunk 4 above)
Comment (rebchua39) [reply src/main/java/seedu/duke/Duke.java:30]: would it be better to use `getMessage()` instead?

Comment (ptejasv) [reply src/main/java/seedu/duke/Duke.java:31]: yep I'm leaving it as printStackTrace just until we have something better to print out the error

Comment (richwill28) [reply src/main/java/seedu/duke/Duke.java:31]: Currently it's like [this](https://github.com/AY2122S1-CS2113T-W11-3/tp/blob/259469b72ea870b4de39afa44ec5fe2c810d6486/src/main/java/seedu/duke/Duke.java#L32). I think it's better if the printing is done inside the `Ui` class instead.

File: src/main/java/seedu/duke/exception/DukeException.java [hunk 6]
```diff
@@ -1,15 +1,16 @@
...
+        super(message);
+    }
+
+    @Override
+    public String toString() {
+        String message = getLocalizedMessage();
+        return (message != null) ? message : Message.ERROR_UNKNOWN;
+    }
+}
+
```
Comment (richwill28) [inline src/main/java/seedu/duke/exception/DukeException.java:16]: An extra blank line, might be a Git coding convention violation.

**Review by richwill28 — APPROVED** [review]
LGTM. Please try to update `text-ui-test` for every PR, just to make sure everything works.  We will be developing JUnit later on to do more comprehensive regression testing.

**Pull request merged**